fuse_models/src/omnidirectional_3d.cpp

```diff
@@ -0,0 +1,629 @@
+ *
+ *  Copyright (c) 2023, Giacomo Franchini
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <Eigen/Dense>
+#include <tf2/utils.h>
+
+#include <stdexcept>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <fuse_core/async_motion_model.hpp>
+#include <fuse_core/constraint.hpp>
+#include <fuse_core/parameter.hpp>
+#include <fuse_core/transaction.hpp>
+#include <fuse_core/uuid.hpp>
+#include <fuse_core/variable.hpp>
+#include <fuse_models/common/sensor_proc.hpp>
+#include <fuse_models/parameters/parameter_base.hpp>
+#include <fuse_models/omnidirectional_3d.hpp>
+#include <fuse_models/omnidirectional_3d_predict.hpp>
+#include <fuse_models/omnidirectional_3d_state_kinematic_constraint.hpp>
+#include <fuse_variables/acceleration_linear_3d_stamped.hpp>
+#include <fuse_variables/orientation_3d_stamped.hpp>
+#include <fuse_variables/position_3d_stamped.hpp>
+#include <fuse_variables/stamped.hpp>
+#include <fuse_variables/velocity_angular_3d_stamped.hpp>
+#include <fuse_variables/velocity_linear_3d_stamped.hpp>
+#include <pluginlib/class_list_macros.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+// Register this motion model with ROS as a plugin.
+PLUGINLIB_EXPORT_CLASS(fuse_models::Omnidirectional3D, fuse_core::MotionModel)
+
+namespace std
+{
+
+inline bool isfinite(const fuse_core::Vector3d & vector)
+{
+  return std::isfinite(vector.x()) && std::isfinite(vector.y() && std::isfinite(vector.z()));
```

Comment:
```suggestion
  return std::isfinite(vector.x()) && std::isfinite(vector.y()) && std::isfinite(vector.z());
```

I think this is a typo